include/robot_localization/navsat_conversions.h

```diff
@@ -135,232 +138,69 @@ static inline void UTM(double lat, double lon, double *x, double *y)
   return;
 }
 
-
 /**
- * Determine the correct UTM letter designator for the
- * given latitude
- *
- * @returns 'Z' if latitude is outside the UTM limits of 84N to 80S
- *
- * Written by Chuck Gantz- chuck.gantz@globalstar.com
- */
-static inline char UTMLetterDesignator(double Lat)
-{
-  char LetterDesignator;
-
-  if     ((84 >= Lat) && (Lat >= 72))  LetterDesignator = 'X';
-  else if ((72 > Lat) && (Lat >= 64))  LetterDesignator = 'W';
-  else if ((64 > Lat) && (Lat >= 56))  LetterDesignator = 'V';
-  else if ((56 > Lat) && (Lat >= 48))  LetterDesignator = 'U';
-  else if ((48 > Lat) && (Lat >= 40))  LetterDesignator = 'T';
-  else if ((40 > Lat) && (Lat >= 32))  LetterDesignator = 'S';
-  else if ((32 > Lat) && (Lat >= 24))  LetterDesignator = 'R';
-  else if ((24 > Lat) && (Lat >= 16))  LetterDesignator = 'Q';
-  else if ((16 > Lat) && (Lat >= 8))   LetterDesignator = 'P';
-  else if (( 8 > Lat) && (Lat >= 0))   LetterDesignator = 'N';
-  else if (( 0 > Lat) && (Lat >= -8))  LetterDesignator = 'M';
-  else if ((-8 > Lat) && (Lat >= -16)) LetterDesignator = 'L';
-  else if ((-16 > Lat) && (Lat >= -24)) LetterDesignator = 'K';
-  else if ((-24 > Lat) && (Lat >= -32)) LetterDesignator = 'J';
-  else if ((-32 > Lat) && (Lat >= -40)) LetterDesignator = 'H';
-  else if ((-40 > Lat) && (Lat >= -48)) LetterDesignator = 'G';
-  else if ((-48 > Lat) && (Lat >= -56)) LetterDesignator = 'F';
-  else if ((-56 > Lat) && (Lat >= -64)) LetterDesignator = 'E';
-  else if ((-64 > Lat) && (Lat >= -72)) LetterDesignator = 'D';
-  else if ((-72 > Lat) && (Lat >= -80)) LetterDesignator = 'C';
-        // 'Z' is an error flag, the Latitude is outside the UTM limits
-  else LetterDesignator = 'Z';
-  return LetterDesignator;
-}
-
-/**
- * Convert lat/long to UTM coords.  Equations from USGS Bulletin 1532
+ * Convert lat/long to UTM coords.
```

Comment:
Can we also get rid of all of this? Or do we still need it?

https://github.com/cra-ros-pkg/robot_localization/blob/ab6e85139d42989e6b988f61fa19aa9583b72afd/include/robot_localization/navsat_conversions.h#L65-L139

And if we do get rid of that, can we also get rid of the other comments and update the license?

https://github.com/cra-ros-pkg/robot_localization/blob/ab6e85139d42989e6b988f61fa19aa9583b72afd/include/robot_localization/navsat_conversions.h#L65-L139

https://github.com/cra-ros-pkg/robot_localization/blob/ab6e85139d42989e6b988f61fa19aa9583b72afd/include/robot_localization/navsat_conversions.h#L65-L139

Reply:
We can.
I left them in because they are exposed to other packages. So I'm not sure who might be using these variables...

Reply:
That's a good point. I'll bring this forward to the ROS2 branches, and for Galactic, I'll remove them. Thanks.

Reply:
I think the whole conversions file can be removed in that case. Since there is now a library doing all that.

Reply:
Agreed.

Reply:
@ayrton04 do you still plan for removing these (quite empty) functions for ROS2 and galactic?

Seems like they are re-used: https://github.com/cra-ros-pkg/robot_localization/pull/676
Which is a bit of a waste.